### ISSUE TITLE
TypeAdapter deserialization failing

### ISSUE THREAD

**Issue by bkchr** [cbeust/klaxon#229]
Hi,
I have the following type declaration:
```
    @TypeFor(field = "type", adapter = SettingValueAdapter::class)
    open class SettingValue(val type: String)
    data class MonitoringTime(val value: Int) : SettingValue("MonitoringTime")
    data class Threshold(val value: Double) : SettingValue("Threshold")

    class SettingValueAdapter : TypeAdapter<SettingValue> {
        override fun classFor(type: Any): KClass<out SettingValue> = when (type as String) {
            "MonitoringTime" -> MonitoringTime::class
            "Threshold" -> Threshold::class
            else -> throw IllegalArgumentException("Unknown type: $type")
        }
    }
```

When I try to decode the following json:
`{"type":"Threshold","value":0.4}`

I get the following exception:
`com.beust.klaxon.KlaxonException: Unable to instantiate Threshold with parameters []`

Any idea what I'm doing wrong?

**Comment by cbeust** [comment]
Looks like a bug, I'll take a look.

**Comment by cbeust** [comment]
Can you post your `SettingValueAdapter`?

**Comment by cbeust** [comment]
Fixed in 5.0.3.

**Comment by bkchr** [comment]
Ty for the fast fix :)
